docs/source/configuration/validation.md

```diff
@@ -0,0 +1,297 @@
+      type: 'integer',
+      maximum: 30
+    },
+  },
+  required: [],
+};
+```
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'maximum',
+  dependencies: {
+    fieldType: 'integer',
+  },
+  method: maximumValidator,
+})
+```
+
+
+#### Field `widget` validators
+
+Field `widget` validators are applied depending on the specified `widget` of the field.
+You should specify the `widget` either in the JSON schema of the block or as additional data in the content type definition.
+
+The following example shows how to specify the `widget` either in the JSON schema of the block.
+
+```ts
+let blockSchema = {
+  // ... fieldset definition in here
+  properties: {
+    ...schema.properties,
+    customField: {
+      title: 'My custom field',
+      description: '',
+      widget: 'phoneNumber',
+    },
+  },
+  required: [],
+};
+```
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'phoneNumber',
+  dependencies: {
+    widgetName: 'phoneNumber',
+  },
+  method: phoneValidator,
+})
+```
+
+The following example shows how to specify the `widget` in the content type definition in the schema hints in the backend using `frontendOptions`.
+The validation engine  will behave the same, applying the `urlValidator` method validator for the content type field `customField` in the previous example.
+
+```python
+from plone.supermodel import model
+from zope import schema
+
+class IMyContent(model.Schema):
+    directives.widget(
+        "customField",
+        frontendOptions={
+            "widget": "url",
+        },
+    )
+    customField = schema.TextLine(
+        title="Custom URL field",
+        required=False,
+    )
+    # Rest of your content type definition
+```
+
+
+#### Behavior and field name validators
+
+Behavior and field name validators are applied depending on the behavior in combination with the name of the field.
+These usually come from a content type definition.
+This type of validator only applies to content type validators.
+It takes the `behaviorName` and the `fieldName` as dependencies.
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'dateRange',
+  dependencies: {
+    behaviorName: 'plone.eventbasic',
+    fieldName: 'start'
+  },
+  method: startEventDateRangeValidator,
+})
+```
+
+
+#### Block type and field name validators
+
+Block type and field name validators are applied depending on the block type in combination with the name of the field in the block settings JSON schema.
+This type of validator only applies to blocks.
+It takes the `blockType` and the `fieldName` as dependencies.
+
+```ts
+config.registerUtility({
+  type: 'validator',
+  name: 'url',
+  dependencies: {
+    blockType: 'slider',
+    fieldName: 'url'
+  },
+  method: urlValidator,
+})
+```
+
+
+(voltos-default-validators-label)=
+
+## Volto's default validators
+
+Volto provides a set of validators by default.
+You can find them in the module {file}`packages/volto/src/config/validators.ts`.
```

Comment:
We should ideally document them here rather than just pointing to source code.

Reply:
Done.